src/uct/sm/self/self.c

```diff
@@ -347,12 +347,12 @@ static ucs_status_t uct_self_md_open(uct_component_t *component, const char *md_
                                      const uct_md_config_t *config, uct_md_h *md_p)
 {
     static uct_md_ops_t md_ops = {
-        .close              = (void*)ucs_empty_function,
+        .close              = ucs_empty_function,
         .query              = uct_self_md_query,
         .mkey_pack          = ucs_empty_function_return_success,
         .mem_reg            = uct_self_mem_reg,
         .mem_dereg          = ucs_empty_function_return_success,
-        .detect_memory_type = (void *)ucs_empty_function_return_zero,
+        .detect_memory_type = ucs_empty_function_return_success
```

Comment:
should be ucs_empty_function_return_unsupported

Reply:
done